File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/Sentry_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class Sentry_Model extends MonsterInfo{
    public static Id = "Sentry"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(39, 45);
        else
            return Mathf.RandomRange(38, 42);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/sentry/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/sentry/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/sentry/skeleton.png"));
    }

    get Name(): string {
        return "哨卫";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/Sentry_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class Sentry_Model : MonsterInfo
    {
        public static string Id = "Sentry";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(39, 45) : Random.Range(38,42);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/sentry/skeleton.atlas",
                    "Monsters/theBottom/sentry/skeleton.json", "Monsters/theBottom/sentry/skeleton.png");
            }
        }

        public override string Name => "哨卫";

        public Sentry_Model(Vector2 pos) : base(pos)
        {
        }
    }
}